app/uk/gov/hmrc/agentsexternalstubs/models/Generator.scala

```diff
@@ -113,6 +113,10 @@ object Generator extends Names with Temporal with Companies with Addresses {
   lazy val crnGen: Gen[String] = pattern"ZZ999999".gen
   def crn(seed: String): String = crnGen.seeded(seed).get
 
+  lazy val cgtPdRefGen: Gen[String] = pattern"999999999".gen.map("XMCGTP" + _)
```

Comment:
I know this a stub, but can we please document any "magic" strings or numbers.  What is the significance of "XMCGTP"?  Why this exact value?  Even better - a one line comment saying why what this produces and why - cheers!

Reply:
CGT ref follows the regex: `X[A-Z]CGTP[0-9]{9}` so we are generating a random CGT ref here... 
first generating a nine-digit number then appending to `XMCGTP`.
if we have to add doc, we have to add for every id like, ARN, UTR.. etc,  documentation is of little value here...

Reply:
OK - well, I find such documentation often useful.  This information is currently in people's heads and if/when people leave then the information is lost.  Imagine if you pick up this code afresh, you will have to ask around and maybe no-one will remember.  I have worked on so many legacy projects where this happens, and I am trying to ensure that future developers have an easier time.  We have a choice - either leave comments out and let people figure it out for themselves or give them a hint.  We can leave this one, but I still want us to either make the code self-documenting or add some comments.  Sorry, as tech lead I have to take responsibility for the code we produce and have inherited.

Reply:
I do not have any problem introducing the doc, but soon we will end up with lots of docs everywhere. our code should be self-documenting... 

the issue above is not the doc, but lack of understanding what cgt ref is,  all our mtd identifiers come from `agent-mtd-identifiers` and they have the regex requirements in there.... and the stub just follows them here...look at how many `Gen`s like this above :) 

regarding the id value, no one knows what those chars represent, for example, we don't know what `X` means and why a CGT ref should start with `X`, we just follow the regex given by DES..

if we have to add the doc, and if i use the below line in `10` services
`lazy val cgtPdRefGen: Gen[String] = pattern"999999999".gen.map("XMCGTP" + _) 
` then we endup adding doc at `10` places to clarify what `XMCGTP` means .. ??

I am not asking people to figure out themselves, but i expect them to have an understanding of what a CGT ref is first, then it is easy for them to figure out what's going on above...

Reply:
it's always better not to have a doc instead of having a confusing or outdated doc
 myself definitely add doc and have always added when the code is particularly tricky

Reply:
Point taken :) Then how about we just have a link to the definition of what CGT Ref is / the pattern / how it is defined?